lib/spack/spack/installer.py

```diff
@@ -1260,6 +1260,10 @@ def _install_task(self, task):
         if not pkg.unit_test_check():
             return
 
+        # Injecting information to know if this installation request is the root one
+        # to determine in BuildProcessInstaller whether installation is explicit or not
+        install_args["is_root"] = task.is_root
```

Comment:
I am not sure of this part, but I found that adding `explicit` argument down the `spack.build_environment.start_build_process` way would imply too much changes.

Reply:
@haampie `explicit` prop is about the overall installation, not the specific one treated by the `start_build_process`. So when installing an implicit dependency, `install_args["explicit"]` will be `True` as the installation of the dependent module has been asked by user.

We need to mix this `explicit` information to the `is_root` one (is this the software asked by user or one of its dependency) to get the accurate implicit state.

Reply:
Right. I meant the task's prop, which is defined as `self.is_root and self.request.install_args.get("explicit", True)` (but not available in `build_process`.

I would personally not duplicate this logic in different places but consolidate ahead of time. So, can't you update `args_for_build_process["explicit"] = task.explicit`?

Reply:
I'm sorry you have to deal with this :laughing: looking at the code it's generally a PITA. It was a mistake to pass effectively command line arguments all the way to an individual package install, instead of transforming them s.t. things like `explicit` were package specific.

Reply:
Do you mean in `def _install_task` to update `install_args["explicit"]` to make it equal to `task.explicit`?

I added another item to the dict (`is_root`) in my initial code, as I thought there may be some underlying part requiring this `explicit` item to be at its initial value.

But if this is the way you suggest, no problem I will update the PR to follow your suggested approach.

Reply:
Thinking about it more the PR is probably good as is. I didn't realize every task holds a reference to the same `install_args` has map, so mutating it is a bad idea, and changing that is maybe beyond this PR.